src/metadataGeneration/controllerGenerator.ts

```diff
@@ -63,6 +66,29 @@ export class ControllerGenerator {
     return decoratorArgument ? `${decoratorArgument.text}` : '';
   }
 
+  private getCommonResponses(): Tsoa.Response[] {
+    const decorators = getDecorators(this.node, identifier => identifier.text === 'Response');
+    if (!decorators || !decorators.length) {
+      return [];
+    }
+
+    return decorators.map(decorator => {
+      const expression = decorator.parent as ts.CallExpression;
```

Comment:
I'd prefer if we would check here, not cast. Something roughly like:
```suggestion
      const expression = decorator.parent;
      
      if (!ts.isCallExpression(expression)) {
        throw new GenerateMetadataError(`The @Response decorator should only be applied at the method or class level`);
      }
```

Reply:
Hm it is not a big deal to fix it, but i don't understand how that case may happens? In `getCommonResponses` method i have check only controller's (`node`'s) decorators, so it is impossible that it may be method's decorators or any other thing. Is it really need to be fixed?